cos-fleetshard-operator/src/main/java/org/bf2/cos/fleetshard/operator/connector/ConnectorConfiguration.java

```diff
@@ -75,15 +75,15 @@ public ConnectorConfiguration(ObjectNode connectorSpec, Class<S> type) {
             }
             this.connectorSpec = (S) result;
         } else {
-            this.connectorSpec = Serialization.unmarshal(connectorSpec.toString(), type);
+            this.connectorSpec = Serialization.unmarshal(connectorSpec.toString(), connectorSpecType);
```

Comment:
I think this `connectorSpec.toString()` can be avoided, given that is is already an ObjectNode it should be possible to use the object mapper to convert it to the target type without the need to go through a string before.

Reply:
Agree! We don't have to de-serialize YAML here (which is covered by `Serialization.unmarshal` too)

Reply:
This applies to `dataShapeSpec` above too.